### PR TITLE
Add encodings.utf_16 stubs

### PR DESCRIPTION
**Pull request by jpy-git**
This PR adds the stubs for the `encodings.utf_16` module in the stdlib.

### CODE REVIEW

File: stdlib/encodings/utf_16.pyi [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    def __init__(self, stream: IO[bytes], errors: str = ...) -> None: ...
+    def reset(self) -> None: ...
+    def encode(self, input: str, errors: str | None = ...) -> tuple[bytes, int]: ...
+
+class StreamReader(codecs.StreamReader):
+    def reset(self) -> None: ...
+    def decode(self, input: bytes, errors: str | None = ...) -> tuple[str, int]: ...
+
+def getregentry() -> codecs.CodecInfo: ...
+def encode(__str: str, __errors: str | None = ..., __byteorder: int = ...) -> tuple[bytes, int]: ...
```
Comment (jpy-git) [inline stdlib/encodings/utf_16.pyi:28]: Quick question here for my understanding as this is my first time typing from one of the C functions:

I know the parameters/types for `codecs.utf_16_encode` by looking at the C code [here](https://github.com/python/cpython/blob/078abb676cf759b1e960f78390b6e80f256f0255/Modules/_codecsmodule.c#L703).

Looking at the similar typing in `utf_8.py` I've prefaced the parameters with a `__` to be consistent. Is that standard practice for all of the C functions?

**Comment by Akuli** [comment]
Do you really need to do `from encodings import utf_16`? There are many encodings which mostly follow the same API, and we shouldn't add stubs for each encoding just for the sake of completeness.

File: stdlib/encodings/utf_16.pyi (hunk 1 above)
Comment (Akuli) [reply stdlib/encodings/utf_16.pyi:28]: `__str` means that the argument is positional-only, meaning that you can't do this:

```
>>> from encodings.utf_16 import encode
>>> encode(str="lol")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: _codecs.utf_16_encode() takes no keyword arguments
```

**Comment by AlexWaygood** [comment]
Just spitballing here:

Maybe one solution would be to have a fictitious `encodings` submodule in `_typeshed`, and then do `from _typeshed.encodings import *` in each of `encodings.utf_8_sig`, `encodings.utf_16`, etc. That way we'd get completeness but wouldn't have to duplicate code everywhere for what's essentially the same API.

**Comment by jpy-git** [comment]
happy to close or do the `_typeshed` suggestion so lmk which you prefer

**Comment by Akuli** [comment]
I think we should just close this. There's already a `__getattr__` in `encodings/__init__.pyi` which prevents getting errors if you import one of these submodules. If someone needs better types, we can add them later, but it's fine to have them available for a couple most common submodules (utf_8 and utf_8_sig).

**Comment by jpy-git** [comment]
sounds good

**Pull request closed without merging**